src/views/Profile/AdminTable/AdminDatePicker.tsx

```diff
@@ -32,18 +35,100 @@ interface IMaterialsDate {
   setChanges: (...params: any[]) => void;
 }
 
+interface ISelectedDay {
+  position: string;
+  bgColor: string;
+  value: string;
+}
+
+const datePickerReducer = (
+  state: {
+    firstSelectedDay: ISelectedDay;
+    secondSelectedDay: ISelectedDay;
+  },
+  action: { type: string; payload: ISelectedDay },
+): {
+  firstSelectedDay: ISelectedDay;
+  secondSelectedDay: ISelectedDay;
+} => {
+  switch (action.type) {
+    case 'SET_SELECTED_DAY':
+      if (action.payload.position === 'start') {
+        return {
+          ...state,
+          firstSelectedDay: {
+            ...state.firstSelectedDay,
+            bgColor: action.payload.bgColor,
+            value: action.payload.value,
+          },
+        };
+      }
+      if (action.payload.position === 'end') {
+        return {
+          ...state,
+          secondSelectedDay: {
+            ...state.secondSelectedDay,
+            bgColor: action.payload.bgColor,
+            value: action.payload.value,
+          },
+        };
+      }
+      break;
+    default:
+      return state;
+  }
+  return state;
+};
+
 export const AdminDatePicker: React.FC<IMaterialsDate> = ({
   start,
   end,
   setChanges,
 }) => {
+  const [datePickerState, dispatch] = useReducer(datePickerReducer, {
+    firstSelectedDay: { position: 'start', bgColor: 'transparent', value: '' },
```

Comment:
am I right that `firstSelectedDay` is always `position: 'start'` when `secondSelectedDay` is always `position: 'end'`?

Reply:
Yes, you are right. I'll change it according to what you wrote above.